protocol/requests/request_to_join_community.go

```diff
@@ -3,31 +3,59 @@ package requests
 import (
 	"errors"
 
+	"github.com/status-im/status-go/eth-node/crypto"
 	"github.com/status-im/status-go/eth-node/types"
 )
 
 var ErrRequestToJoinCommunityInvalidCommunityID = errors.New("request-to-join-community: invalid community id")
+var ErrRequestToJoinCommunityNoAddressesToReveal = errors.New("request-to-join-community: no addresses to reveal")
 var ErrRequestToJoinCommunityMissingPassword = errors.New("request-to-join-community: password is necessary when sending a list of addresses")
 var ErrRequestToJoinNoAirdropAddress = errors.New("request-to-join-community: airdropAddress is necessary when sending a list of addresses")
+var ErrRequestToJoinNoAirdropAddressAmongAddressesToReveal = errors.New("request-to-join-community: airdropAddress must be in the set of addresses to reveal")
+var ErrRequestToJoinCommunityInvalidSignature = errors.New("request-to-join-community: invalid signature")
 
 type RequestToJoinCommunity struct {
-	CommunityID       types.HexBytes `json:"communityId"`
-	ENSName           string         `json:"ensName"`
-	Password          string         `json:"password"`
-	AddressesToReveal []string       `json:"addressesToReveal"`
-	AirdropAddress    string         `json:"airdropAddress"`
+	CommunityID       types.HexBytes   `json:"communityId"`
+	ENSName           string           `json:"ensName"`
+	AddressesToReveal []string         `json:"addressesToReveal"`
+	Signatures        []types.HexBytes `json:"signatures"` // the order of signatures should match the order of addresses
```

Comment:
can you add the signature to the database? I must have the possibility to re-send existing user requests to join with revealed accounts without signing in it again

Reply:
@mprakhov that can be done as a new issue, especially cause I don't see that we have resend mechanics developed yet.